### ISSUE TITLE
Pyright create var is possiply unbound warning with arrays

### ISSUE THREAD

**Issue by realhackcraft** [microsoft/pyright#7710]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
I'm not sure what this feature is called in python, but when I type the following code (which runs), Pyright is giving me possiply unbound warnings.

**Code or Screenshots**
```python
list = ["a", "b", "c"]
a, b, c = list

# a, b, and c is possibly unbound
```
**VS Code extension or command-line**
Nvim

**Comment by erictraut** [comment]
I'm not able to repro the issue you're reporting. Can you please confirm that when you paste the above code into your editor, pyright is reporting an error? Perhaps you have some other language server or linter installed as well?

Here is your code pasted into the [pyright playground](https://pyright-play.net/?code=DYSwzgLgBAvFDaAiAhogNFRAjdmDGiAugFDIZYZ6xSiTHFA).

**Comment by realhackcraft** [comment]
Nevermind... I found that in my code I declared the varables one scope too deep.

Adding a = b = ""

before the indented code fixed it.

I'm a bit new to python so still haven't figured out the ropes yet, sorry.
